src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx

```diff
@@ -0,0 +1,369 @@
+            return configData
+        }
+
+        return null
+    }, [comparisonDataLoader, comparisonData, sortOrder])
+
+    // SELECT PICKER OPTIONS
+    /** Compare Environment Select Picker Options  */
+    const compareEnvironmentSelectorOptions: OptionsOrGroups<
+        SelectPickerOptionType,
+        GroupBase<SelectPickerOptionType>
+    > = [
+        {
+            label: BASE_CONFIGURATIONS.name,
+            value: '',
+        },
+        {
+            label: 'Environments',
+            options: environments.map(({ name }) => ({
+                label: name,
+                value: name,
+            })),
+        },
+        {
+            label: 'Default values',
+            options:
+                currentEnvOptions?.defaultVersions.map(({ chartRefId: _chartRefId, chartVersion }) => ({
+                    label: `v${chartVersion} (Default)`,
+                    value: _chartRefId,
+                })) || [],
+        },
+    ]
+
+    /**
+     * Get the configuration type options for the current or compare environment.
+     *
+     * @param isCompare - If true, compare env data will be used.
+     * @returns The configuration type options for the select picker.
+     */
+    const getEnvironmentConfigTypeOptions = (
+        isCompare?: boolean,
+    ): OptionsOrGroups<SelectPickerOptionType, GroupBase<SelectPickerOptionType>> => [
+        {
+            label: 'Published only',
+            value: AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+            description: 'Configurations that will be deployed in the next deployment',
+        },
+        {
+            label: 'Drafts only',
+            value: AppEnvDeploymentConfigType.DRAFT_ONLY,
+            description: 'Configurations in draft or approval pending state',
+        },
+        {
+            label: 'Saved (Includes drafts)',
+            value: AppEnvDeploymentConfigType.PUBLISHED_WITH_DRAFT,
+            description: 'Last saved configurations including any drafts',
+        },
+        {
+            label: 'Previous Deployments',
+            options: ((isCompare ? compareEnvOptions : currentEnvOptions)?.previousDeployments || []).map(
+                ({ finishedOn, chartVersion, pipelineId: _pipelineId, deploymentTemplateHistoryId }) => ({
+                    label: `${moment(finishedOn).format(Moment12HourFormat)} (v${chartVersion})`,
+                    value: getPreviousDeploymentOptionValue(deploymentTemplateHistoryId, _pipelineId),
+                }),
+            ),
+        },
+    ]
+
+    // METHODS
+    const onCompareEnvironmentChange = ({ value }: SelectPickerOptionType) => {
+        if (typeof value === 'number') {
+            updateSearchParams({
+                chartRefId: value,
+                compareWithConfigType: AppEnvDeploymentConfigType.DEFAULT_VERSION,
+                compareWithIdentifierId: null,
+                compareWithPipelineId: null,
+            })
+        } else if (typeof value === 'string') {
```

Comment:
we can add a comment as to what does this number/string specify

Reply:
yes will add comments for whole file separately